Automattic/jetpack
# exploring dedicated request

PROOF OF CONCEPT

This is an exploration PR not meant to be merged at any point.

It's just an annotated patch with my exploration in the last two days, to serve as a starting point for other people working on this project or to myself when I'm back from holidays.

Look at the changes and comments.

In order to test:

* you need to run `\Automattic\Jetpack\Sync\Settings::update_settings( [ 'dedicated_request_enable' => true ] );` in order to set the feature flag (you'll see an error if you don't do it)
* Import some content to your site as a way to create a long queue
* Make sure to close all browser windows to not have any requests being made that would process the queue and confuse things
* tail -f your webserver error log
* make a simple request to the site's home page (a wget will do it)
* watch the logs and the queue size
* you can also sandbox and watch the sync actions being triggered on your sandbox
